RenderSystems/GLSupport/src/EGL/Wayland/OgreWaylandEGLWindow.cpp

```diff
@@ -0,0 +1,381 @@
+}
+
+void WaylandEGLWindow::windowMovedOrResized()
+{
+    if (mClosed || !mWindow)
+        return;
+
+    int width, height;
+    wl_egl_window_get_attached_size(mWindow, &width, &height);
+    resize(width, height);
+}
+
+void WaylandEGLWindow::switchFullScreen(bool fullscreen)
+{
+    if (mXdgToplevel)
+    {
+        if (fullscreen)
+            xdg_toplevel_set_fullscreen(mXdgToplevel, nullptr);
+        else
+            xdg_toplevel_unset_fullscreen(mXdgToplevel);
+
+        wl_surface_damage(mGLSupport->mWlSurface, 0, 0, mWidth, mHeight);
+        wl_surface_commit(mGLSupport->mWlSurface);
+        wl_display_dispatch_pending(mNativeDisplay);
+        wl_display_flush(mNativeDisplay);
+    }
+}
+
+void WaylandEGLWindow::create(const String& name, uint width, uint height, bool fullScreen,
+                              const NameValuePairList* miscParams)
+{
+    String title = name;
+    int samples = 0;
+    short frequency = 0;
+    bool vsync = false;
+    ::EGLContext eglContext = NULL;
+
+    unsigned int vsyncInterval = 1;
+
+    //mNativeDisplay = mGLSupport->getNativeDisplay();
+    mIsFullScreen = fullScreen;
+
+    if (miscParams)
+    {
+        NameValuePairList::const_iterator opt;
+        NameValuePairList::const_iterator end = miscParams->end();
+
+        if ((opt = miscParams->find("currentGLContext")) != end && StringConverter::parseBool(opt->second))
+        {
+            eglContext = eglGetCurrentContext();
+            EGL_CHECK_ERROR
+            if (!eglContext)
+            {
+                OGRE_EXCEPT(Exception::ERR_RENDERINGAPI_ERROR,
+                            "currentGLContext was specified with no current GL context", "EGLWindow::create");
+            }
+
+            mEglDisplay = eglGetCurrentDisplay();
+            EGL_CHECK_ERROR
+        }
+
+        if ((opt = miscParams->find("FSAA")) != end)
+        {
+            samples = StringConverter::parseUnsignedInt(opt->second);
+        }
+
+        if ((opt = miscParams->find("displayFrequency")) != end)
+        {
+            frequency = (short)StringConverter::parseInt(opt->second);
+        }
+
+        if ((opt = miscParams->find("vsync")) != end)
+        {
+            vsync = StringConverter::parseBool(opt->second);
+        }
+
+        if ((opt = miscParams->find("vsyncInterval")) != end)
+            vsyncInterval = StringConverter::parseUnsignedInt(opt->second);
+
+        if ((opt = miscParams->find("gamma")) != end)
+        {
+            mHwGamma = StringConverter::parseBool(opt->second);
+        }
+
+        if ((opt = miscParams->find("title")) != end)
+        {
+            title = opt->second;
+        }
+
+        if ((opt = miscParams->find("externalGLControl")) != end)
+        {
+            mIsExternalGLControl = StringConverter::parseBool(opt->second);
+        }
+    }
+
+    initNativeCreatedWindow(miscParams);
+
+    mGLSupport->doInit();
+
+
+    if (!mEglConfig)
+    {
+        int minAttribs[] = {EGL_RED_SIZE,
```

Comment:
it should be sufficient to specify those:
https://github.com/OGRECave/ogre/blob/b9fbf7da95a9de887944de0633ea77983988c375/RenderSystems/GLSupport/src/EGL/Emscripten/OgreEmscriptenEGLWindow.cpp#L318-L334

Reply:
I added your suggestion. Your suggestion refer to some member attributes, e.g. mMaxBufferSize. I added these as variables local to the function.